bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerDeviceHandler.java

```diff
@@ -36,19 +41,24 @@ public OpenSprinklerDeviceHandler(Thing thing) {
 
     @Override
     protected void updateChannel(ChannelUID channel) {
-        switch (channel.getIdWithoutGroup()) {
-            case SENSOR_RAIN:
-                try {
+        try {
+            switch (channel.getIdWithoutGroup()) {
+                case SENSOR_RAIN:
                     if (getApi().isRainDetected()) {
                         updateState(channel, OnOffType.ON);
                     } else {
                         updateState(channel, OnOffType.OFF);
                     }
-                } catch (CommunicationApiException e) {
-                    logger.debug("Could not update rainsensor", e);
-                }
-            default:
-                logger.debug("Not updating unknown channel {}", channel);
+                    break;
+                case SENSOR_CURRENT_DRAW:
+                    updateState(channel, new QuantityType<ElectricCurrent>(new Integer(getApi().currentDraw()),
```

Comment:
Primitives should be automatically boxed for you.
```suggestion
                    updateState(channel, new QuantityType<ElectricCurrent>(getApi().currentDraw(),
```